### DISCUSSION TITLE
Make single-file copy / cut & paste select the new file created

### DISCUSSION THREAD

**Discussion by doronbehar** [gokcehan/lf#2247] [Q&A]
Say you select a single file for a copy / move. You then paste it in a different directory. I think it's intuitive, especially if it's a large directory, that the new file pasted will be selected with `select`.

This can be overridden by the following:

```
# paste and select the file pasted
# paste and select the file pasted
cmd paste-and-select %{{
	lf -remote 'load' | while read file; do
		if [ ! -n "${mode+1}" ]; then
			mode="$file"
			continue
		fi
		if [ -f "${file}" ]; then
			# get 'basename' of the selection
			filename="${file##*/}"
			# quote it so we won't deal with quotes in the lf -remote command
			filename="$(printf '%q' "$filename")"
			break
		fi
	done
	lf -remote "send $id paste"
	lf -remote "send $id select $filename"
}}
map p paste-and-select
```

Should this behavior be the default? Open question, I tend to think it should, but my opinion is not strong.

**Comment by gokcehan** [comment]
@doronbehar I don't have a strong opinion as well but we should also consider what should happen when multiple files/dirs are copied for this case.

**Comment by doronbehar** [comment]
In the current implementation of that config snippet, and in the ideal (IMO) behavior of this, the first file that was loaded is selected in the target directory.

**Comment by doronbehar** [comment]
BTW for the record, with large files / directories copy-paste, my workaround doesn't work because the `send $id select` command is sent before the paste is finished (it works in the background)... I can workaround that with an ugly `sleep` but that's not a real solution.

**Comment by CatsDeservePets** [comment]
Hello @doronbehar!

Are you still interested in this? 
If so, what are your stands on the following:
- When copying/moving multiple files, which file should be selected?
- If the process of copying a file takes some time and you already moved (either down to another file or even inside another location), what should happen? Should you move back to copied file? Should you not be allowed to move or select another file until the others are copied? Should nothing happen if you already moved the cursor?

**Comment by doronbehar** [comment]
Hello @CatsDeservePets ! Thanks for your interest.

> * When copying/moving multiple files, which file should be selected?

I think that as a start, we can avoid change the behavior when copying/moving multiple files.

> * If the process of copying a file takes some time and you already moved (either down to another file or even inside another location), what should happen? Should you move back to copied file? Should you not be allowed to move or select another file until the others are copied? Should nothing happen if you already moved the cursor?

I think that the best behavior is the following: The `p` key, should simply ask the user in advance - before the 'pasting' actually starts, which file name to use.

**Comment by CatsDeservePets** [comment]
Ok, as this is something really specific, we are no longer talking about a built-int option but a custom command.

You could use this https://github.com/gokcehan/lf/issues/1616 as a starting point. You can find a similar script that acts different whether you are using one or more files.
I hope this helps.

Don't hesitate to ask if you need more help.
